### PR TITLE
Move `Overlay` styles to PVC

### PR DESCRIPTION
**Pull request by simurai**
### Description

This is part of [#1342](https://github.com/github/primer/issues/1342) and on top of https://github.com/primer/view_components/pull/1401. It adds the [`Overlay`](https://github.com/primer/css/blob/aee4b6f571d88f391fcf98170857c4eed7b1ae82/src/overlay/overlay.scss) styles from PCSS. There should be no visual changes.

### Integration

> Does this change require any updates to code in production?

Yes, the [following lines](https://github.com/github/github/blob/985ee396bf03faf4c2cd43d0d6fc1ef9248a46cf/app/assets/stylesheets/bundles/primer/index.scss#L83) can be removed on dotcom:

```diff
- @import '@primer/css/overlay/overlay.scss';
```

### Merge checklist

- [ ] Added/updated tests
- [ ] ~~Added/updated documentation~~
- [ ] Added/updated previews

### CODE REVIEW

File: app/components/primer/alpha/overlay.pcss [hunk 1]
```diff
@@ -17,14 +19,496 @@ primer-overlay[popover='manual' i] {
...
+/* primer-overlay[popover='' i].\:open,
 primer-overlay[popover='auto' i].\:open,
 primer-overlay[popover='manual' i].\:open {
   display: block;
   position: fixed;
   z-index: 2147483647;
-}
+} */
+
 /* stylelint-enable selector-class-pattern */
```
Comment (simurai) [inline app/components/primer/alpha/overlay.pcss:34]: I only commented out these `primer-overlay` styles. It seems to be "test styles", but I assume they can be removed again at some point?

Before | After
--- | ---
Screen Shot 2022-11-29 at 17 56 52 | Screen Shot 2022-11-29 at 17 57 32

**Comment by simurai** [comment]
@keithamus Not sure what the status of https://github.com/primer/view_components/pull/1648 is, but feel free to merge in these styles. Or you can also leave them as a separate PR and we can merge them later.

**Review by keithamus — APPROVED** [review]
Let's get these into main and forgo merging into #1401 for now. I'll rebase #1401 after the fact, as `Dialog` also uses these styles.

**Comment by simurai** [comment]
Superseded by https://github.com/primer/view_components/pull/1651 that is intended to be merged into `main`.

**Pull request closed without merging**